test/controllers/devise_token_auth/sessions_controller_test.rb

```diff
@@ -72,6 +72,23 @@ class DeviseTokenAuth::SessionsControllerTest < ActionController::TestCase
             assert_equal '0.0.0.0', @new_last_sign_in_ip
           end
         end
+
```

Comment:
Shouldn't you assert that the _correct_ tokens (i.e. the _oldest_ tokens) are deleted here?

E.g.: 
 Given that the `max_number_of_devices = 5`, and there are 5 existing tokens,
 When an additional token is added, the oldest token is dropped and the new one is present.